### PR TITLE
rewrite version of the merge script

### PR DESCRIPTION
**Pull request by wjwwood**
As requested. I have commented out all but `moveit_commander` and `moveit_experimental` at the top of the script, since the first is small and the second is a snowflake. I pushed the results to my github:

https://github.com/wjwwood/moveit

I haven't run a full pass just yet because it will take an hour or so. Rewriting the history can take a long time when there are so many commits.

### CODE REVIEW

File: git_rewrite_merge_moveit.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            call(['git', 'branch', '-d', temp_branch])
+            trashed_files = False
+            if os.path.isfile(os.path.join(repo, '.gitignore')):
+                trashed_files = True
+                call(['git', 'rm', os.path.join(repo, '.gitignore')])
+            if os.path.isfile(os.path.join(repo, '.travis.yml')):
+                trashed_files = True
+                call(['git', 'rm', os.path.join(repo, '.travis.yml')])
+            if trashed_files:
+                call(['git', 'commit', '-m', 'removing vestigial files after merging'])
```
Comment (davetcoleman) [inline git_rewrite_merge_moveit.py:111]: "vestigial" nice word!

File: git_rewrite_merge_moveit.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+        print("\n==> Adding new README.md and .gitignore for branch '{0}'".format(branch))
+        template_file(
+            os.path.join(template_dir, 'README.md.template'),
+            os.path.join('README.md'),
+            {
+                '__DISTRO__': distro,
+                '__DISTRO_CAP__': distro.capitalize(),
```
Comment (davetcoleman) [inline git_rewrite_merge_moveit.py]: This didn't seem to work [here](https://github.com/wjwwood/moveit/commit/a517459271f2255df42685a5eed0a350bb3acc13#diff-04c6e90faac2675aa89e2176d2eec7d8R43)

**Comment by davetcoleman** [comment]
Awesome, this looks great! (I didn't except you to rewrite it to python)

Perhaps I can still use the last piece of my bash script that verifies each repository across all branches, as a full proof check of this python version

Also, in computing what do you mean by "snowflake"? :snowflake:

**Comment by wjwwood** [comment]
Yeah, I started out modifying your script, but wanted to preserve it side-by-side, and then ended up getting frustrated without a dictionary, so I rewrote it in Python. It wasn't bad, and it helped me get a better handle on how the first script worked. My script doesn't do everything yours does. It's just a quick and dirty port.

The magic happens around lines 90-101, and in particular the filter-branch command:

https://github.com/davetcoleman/moveit_merge/pull/6/files#diff-4c8a6802846c4cfb9efa17f6d23cdb59R94

**Comment by wjwwood** [comment]
> Also, in computing what do you mean by "snowflake"? ❄️

That it is its own special little situation that needs to be handled with care 😄.

**Comment by davetcoleman** [comment]
Did you run into the issue where you can't merge multiple branches from the original repos into multiple branches in the new repo without the new repo having a unique branch name? That is why at the start I have:

> git checkout -b indigo-devel-temporary-unique-name

**Comment by davetcoleman** [comment]
Also I added error catching to every command to abort the script in case anything goes wrong, that seems important

**Comment by wjwwood** [comment]
My branch names are unique because they are `temp/{repo_name}/{branch_name}`. I also delete them when I'm done with them, but before creating the new branch.

`check_call` will raise an exception if the return code is non-zero.

**Pull request merged**